slides.md

```diff
@@ -17,6 +17,170 @@ class: middle, center
 
 class: middle, center
 
+## The Context API
+
+---
+
+class: middle, center
+
+## The Hooks API
+
+---
+
+class: middle
+
+> Hooks are a new feature proposal that lets you use state and other React features without writing a class. They’re currently in **React v16.7.0-alpha**
+
+---
+
+### What does Hook solve?
+
+React Component is great to organize large UI into small, independent, and reusuable piece.
+
+However, it is quite challenging to break down complex components, since the logic is **stateful** and can't be extracted to a function or another component.
+
+---
+
+### Complex components, with logic all over the lifecyle
+
+```typescript
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { name: "Kawamura", like: "curry" };
+  }
+  componentDidMount() {
+    fetchA();
+    fetchB();
+    subscribe();
+  }
+  componentWillUpdate(props) {
+    if (props.a !== this.props.a) fetchA();
+    if (props.b !== this.props.b) fetchB();
+  }
+  componentWillUnmount() { unSubscribe(); }
+  handleNameChange(event) { this.setState({ name: event.target.value }); }
+  render() { ... }
+}
+```
+
+Each _lifecycle methods_ is consisted with **unrelated logics**, and the related code that changes together gets **split apart**.
```

Comment:
```suggestion
The business logic, in all _lifecycle methods_ are _internally_ **inconsistent**, `componentDidMount` for instance will fetch `A` and `B` here.

Also, if we wanted to extract the methods, the logic would be **split apart**.

Basically, instead of splitting the code for technical reasons, we want to split by concern.
```